backends/gdx-backend-android/src/com/badlogic/gdx/backends/android/AndroidZipFileHandle.java

```diff
@@ -108,25 +108,31 @@ public FileHandle parent() {
 	@Override
 	public FileHandle[] list() {
 		ZipEntryRO[] zipEntries = expansionFile.getEntriesAt(getPath());
-		FileHandle[] handles = new FileHandle[zipEntries.length];
-		for (int i = 0, n = handles.length; i < n; i++)
-			handles[i] = new AndroidZipFileHandle(zipEntries[i].mFileName);
+		FileHandle[] handles = new FileHandle[zipEntries.length - 1];
+		int count = 0;
+		for (int i = 0, n = zipEntries.length; i < n; i++) {
+			if (zipEntries[i].mFileName.length() == getPath().length())
```

Comment:
Why comparing the lengths of the file names?

Reply:
I believe it's a comparison with the directory path. It should always be shorter than the path of any file inside the directory, and I figured comparing the lengths was faster than a string or hash comparison.

Reply:
Ah, missed that it was mFileName is including the full path also. In that case should be good.